### PR TITLE
Cms Blog AdminAppService is refactored

### LINKED ISSUE #7850: Refactor CmsKit Blog Admin AppService

### PR DESCRIPTION
**Pull request by cotur**
Closes: https://github.com/abpframework/abp/issues/7850

### CODE REVIEW

File: modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogAdminAppService.cs [hunk 1]
```diff
@@ -14,24 +12,63 @@ namespace Volo.CmsKit.Admin.Blogs
...
+            await BlogRepository.UpdateAsync(blog);
+
+            return ObjectMapper.Map<Blog, BlogDto>(blog);
         }
 
-        protected override async Task<IQueryable<Blog>> CreateFilteredQueryAsync(BlogGetListInput input)
+        [Authorize(CmsKitAdminPermissions.Blogs.Delete)]
+        public virtual Task DeleteAsync(Guid id)
         {
-            var queryable = await base.CreateFilteredQueryAsync(input);
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogAdminAppService.cs:30]: Should we add a check here? What happens if the blog is not found?

File: modules/cms-kit/src/Volo.CmsKit.Domain.Shared/Volo/CmsKit/CmsKitErrorCodes.cs [hunk 2]
```diff
@@ -16,6 +16,11 @@ public static class Pages
         }
 
         public static class Blogs
+        {
+            public const string SlugAlreadyExist = "CmsKit:Blog:0001";
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Domain.Shared/Volo/CmsKit/CmsKitErrorCodes.cs]: `SlugAlreadyExist`

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/BlogManager.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public class BlogManager : DomainService
+    {
+        protected IBlogRepository BlogRepository { get; }
+        
+        public BlogManager(IBlogRepository blogRepository)
+        {
+            BlogRepository = blogRepository;
+        }
+
+        public virtual async Task<Blog> CreateAsync([NotNull] string name, [NotNull] string slug)
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/BlogManager.cs:16]: we should also accept the tenantId parameter as optional. The application layer should pass the CurrentTenant.Id.

File: modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogAdminAppService.cs (hunk 1 above)
Comment (cotur) [reply modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogAdminAppService.cs:30]: Repository checks the entity before deleting. If there is no entity, it won't cast deleting action.

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/BlogManager.cs (hunk 3 above)
Comment (cotur) [reply modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/BlogManager.cs:16]: By design, we don't want to move tenantId by code.

File: modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/Blog.cs [hunk 4]
```diff
@@ -9,33 +9,29 @@ namespace Volo.CmsKit.Blogs
...
         [NotNull] 
         public virtual string Name { get; protected set; }
 
         [NotNull] 
         public virtual string Slug { get; protected set; }
 
         public virtual Guid? TenantId { get; protected set; }
-
+        
+        internal Blog(Guid id, [NotNull] string name, [NotNull] string slug, [CanBeNull] Guid? tenantId = null) : base(id)
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Blogs/Blog.cs]: maybe we should make it `protected internal`

**Review by ilkayilknur — APPROVED** [review]

**Pull request merged**